### PR TITLE
[Term Entry] Lua Coroutines .status()

### LINKED ISSUE #3382: [Term Entry] Lua Coroutines .status()
### Reason/inspiration (optional)

We would like a new term entry on the `.status()` method for Lua: [Coroutines](https://www.codecademy.com/resources/docs/lua/coroutines). The entry should go in a new file under `docs/content/lua/concepts/coroutines/terms/status/status.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```lua `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by noahpgordon**
### Description

Add term **status.md** to Lua Coroutines. Closes #3382 .

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/coroutines/terms/status/status.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - 'Functions'
+  - 'Async Await'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The method **`.status()`** takes a coroutine as a parameter and returns the status of that coroutine. This status can be one of three states: `suspended`, `running`, or `dead`.
+
+When a coroutine is created, it has a status of `suspended`. Once a coroutine is ran or resumed, it has a status of `running` until it is either re-suspended using `coroutine.yield()`, or it completes. If a coroutine completes its execution and terminates, it receives a permanent status of `dead`.
```
Comment (caupolicandiaz) [inline content/lua/concepts/coroutines/terms/status/status.md:17]:
```suggestion
When a coroutine is created, it has a status of `suspended` by default. Once a coroutine has been called or resumed, it has a status of `running`.  The coroutine will run until it is suspended, using `coroutine.yield()`, or it completes. If a coroutine completes its execution and terminates, it receives a permanent status of `dead`.
```

File: content/lua/concepts/coroutines/terms/status/status.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Syntax
+
+The syntax for calling `.status()` is given below:
```
Comment (caupolicandiaz) [inline content/lua/concepts/coroutines/terms/status/status.md:21]:
```suggestion
The syntax for calling `status()` is given below:
```

File: content/lua/concepts/coroutines/terms/status/status.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+```pseudo
+coroutine.status(parameterCoroutine)
```
Comment (caupolicandiaz) [inline content/lua/concepts/coroutines/terms/status/status.md:24]:
```suggestion
coroutine.status(xCoroutine)
```

File: content/lua/concepts/coroutines/terms/status/status.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+Where `parameterCoroutine` names the coroutine to be checked.
```
Comment (caupolicandiaz) [inline content/lua/concepts/coroutines/terms/status/status.md:27]:
```suggestion
Where `xCoroutine` names the coroutine to be checked.
```

File: content/lua/concepts/coroutines/terms/status/status.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+end
+
+exampleCoroutine = coroutine.create(lifecycle)
+
+print(coroutine.status(exampleCoroutine)) --Prints 'suspended', since we have not yet run exampleCoroutine.
+
+coroutine.resume(exampleCoroutine, exampleCoroutine) --Runs exampleCoroutine and passes it as a parameter to lifecycle(). This prints 'running', since .status() is called on exampleCoroutine while it is executing.
+
+print(coroutine.status(exampleCoroutine)) --Prints 'dead', since exampleCoroutine has terminated.
+```
```
Comment (caupolicandiaz) [inline content/lua/concepts/coroutines/terms/status/status.md:47]: the output should be in a ```shell block instead of as comments

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@noahpgordon I've added my initial comments. Please update the entry, and tag me when you've completed the changes. Thanks

File: content/lua/concepts/coroutines/terms/status/status.md (hunk 5 above)
Comment (noahpgordon) [reply content/lua/concepts/coroutines/terms/status/status.md:47]: I added a shell block but also kept the comments as explanations of the output.

**Comment by noahpgordon** [comment]
@caupolicandiaz I've made my edits.

**Review by caupolicandiaz — APPROVED** [review]
@noahpgordon This is ready for a second review.

**Review by KTom101 — APPROVED** [review]
@noahpgordon I've added a minor change directly and applied formatting. It's good to go.

**Pull request merged**